### ISSUE TITLE
Validation set for hyperparameter adjustment

### ISSUE THREAD

**Issue by anikristo** [CS-464-Ominatori/Ominatori#27]
Train on the training set, adjust hyperparameters for best performance on validation set and only use the test set for measuring the final performance.
